### PR TITLE
[washingtonpost] Add feature to read data from json that will be loaded on page.

### PR DESCRIPTION
**Pull request by Bricio**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

#2778

Add feature to read data from JSON that will be read by page JS.

### CODE REVIEW

File: yt_dlp/extractor/washingtonpost.py [hunk 1]
```diff
@@ -106,6 +118,14 @@ def _real_extract(self, url):
                 <div\s+class="posttv-video-embed[^>]*?data-uuid=|
                 data-video-uuid=
             )"([^"]+)"''', webpage)
+
+        if not uuids:
+            next_data = get_element_by_id('__NEXT_DATA__', webpage)
+            json_data = self._parse_json(next_data, page_id, transform_source=js_to_json)
```
Comment (pukkandan) [inline yt_dlp/extractor/washingtonpost.py:124]: `self._search_nextjs_data`

File: yt_dlp/extractor/washingtonpost.py [hunk 2]
```diff
@@ -59,9 +61,10 @@ class WashingtonPostArticleIE(InfoExtractor):
                 'description': 'Overly complicated paper pushing is nothing new to government bureaucracy. But the way federal retirement applications are filed may be the most outdated. David Fahrenthold explains.',
                 'timestamp': 1395440416,
                 'upload_date': '20140321',
+                'thumbnail': 'https://d1i4t8bqe7zgj6.cloudfront.net/PAPERMINESplash.jpg',
```
Comment (pukkandan) [inline yt_dlp/extractor/washingtonpost.py:64]: The subdomain may change. Use regex

File: yt_dlp/extractor/washingtonpost.py [hunk 3]
```diff
@@ -50,7 +52,7 @@ class WashingtonPostArticleIE(InfoExtractor):
             'title': 'Sinkhole of bureaucracy',
         },
         'playlist': [{
-            'md5': 'b9be794ceb56c7267d410a13f99d801a',
+            'md5': '7ccf53ea8cbb77de5f570242b3b21a59',
```
Comment (pukkandan) [inline yt_dlp/extractor/washingtonpost.py:55]: why are the md5 changing? is this a fragmented download? In that case, md5 check should be removed

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/washingtonpost.py (hunk 3 above)
Comment (Bricio) [reply yt_dlp/extractor/washingtonpost.py:55]: Dont know why it changed. Maybe a new video version on server side?
But it didn't change, after I fixed, in every tests that I made.

I only added code to get playlist item from next_js, if none is found on previous search. 
This 2 test links was working before and are not impacted by the new code. 
I only changed the md5 value to pass through tests.

**Pull request merged**